flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -574,8 +574,11 @@ private static String getEntryPointClassNameFromJar(URL jarFile) throws ProgramI
 	}
 	
 	private static Class<?> loadMainClass(String className, ClassLoader cl) throws ProgramInvocationException {
+		ClassLoader contextCl = null;
 		try {
-			return Class.forName(className, true, cl);
+			contextCl = Thread.currentThread().getContextClassLoader();
+			Thread.currentThread().setContextClassLoader(cl);
+			return Class.forName(className, false, cl);
```

Comment:
Required for the Clojure case. If we initialize right here, the `main` method gets invoked right away. This is too early as not everything got set up properly at this point by the client and the execution fails.